_resources/port1.0/group/gpg_verify-1.0.tcl

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+#
+# This portgroup is for ports that verify package-provided gpg signatures
+#
+# A single signature verification by the Portfile author is sufficient,
+# along with the standard checksum phase. The default Portfile behavior
+# is not to check gpg signatures, and assume that the Portfile author
+# has already done this. The flag `gpg_verify.use_gpg_verification`
+# is used within conditionals to run the procedure `gpg_verify.verify_gpg_signature`.
+# 
+# Usage:
+# PortGroup       gpg_verify 1.0
+# set gpg_verify.use_gpg_verification   true
+# gpg_verify.verify_gpg_signature       keyid_file signature_file test_file
+#
+# Note that many PGP key servers are intermittently accessible; therefore, it is
+# good practice to include the keyid file in ${filespath}. Also, the checksum
+# phase requires at least one hash check of downloaded files, including PGP
+# signatures for which hash checks are unnecessary. It is therefore recommended
+# to use `size` for signature file checksums, which is often constant for the same
+# keyid.
+# Example (from the julia Portfile):
+# if {${use_gpg_verification} == "true"} {
+#     distfiles-append \
+#                     ${name}-${version}-full${extract.suffix}.asc
+#     checksums-append \
+#                     ${name}-${version}-full${extract.suffix}.asc \
+#                     size    866
+# }
+
+set gpg_verify.use_gpg_verification     false
+
+set gpg_verify.gpg_homedir              ${workpath}/.gnupg
+
+proc gpg_verify.verify_gpg_signature {gpg_homedir keyid_file signature_file test_file} {
+    # pre-load public key to avoid keyserver downtime issues
+    # https://pgp.mit.edu/pks/lookup?op=get&search=0x${gpg_keyid}
+    # note: tcl exec will return error if error messages not directed to /dev/null
+    system "/bin/sh -c 'gpg --homedir ${gpg_verify.gpg_homedir} \
+        --import ${keyid_file} 2>/dev/null || /usr/bin/true'"
+    set gpg_verification [exec /bin/sh -c "if gpg --homedir ${gpg_verify.gpg_homedir} \
```

Comment:
This is not strictly using the `gpg` command from MacPorts (i.e. in `${prefix}/bin`). Should using `gpg` outside of MacPorts (e.g. when installed via GPG Suite) be possible? If so, then the `gpg` dependency might be specified where MacPorts' `gpg` is only a fallback, i.e. `bin:gpg:gnupg2` instead of `port:gnupg2`.

Reply:
I'm pretty sure MacPorts sticks `${prefix}/bin` at the top of `PATH`, so you'd get it by default. 

Just saying `gpg` gives the option to alter the path and use a different version if desired.

I'll go with whatever the consensus is. I only use the MacPorts version myself.

Reply:
It also might be possible for this dependency to be specified in the portgroup rather than the individual ports. I'm not sure which is better, but I imagine specifying it in one place is could be better for consistency/ease of use.

Reply:
> I'm pretty sure MacPorts sticks `${prefix}/bin` at the top of `PATH`, so you'd get it by default.

Doesn't this only apply to the user's `PATH`, and not the one used by `port`? See [note](https://guide.macports.org/chunked/installing.shell.html#installing.shell.postflight) in guide:

>### Note
>The user environment's $PATH is not in effect while ports are being installed, because the $PATH is scrubbed before ports are installed, and restored afterwards. To change the search path for locating system executables (rsync, tar, etc.) during port installation, see the macports.conf file variable binpath. But changing this variable is for advanced users only, and is not generally needed or recommended.

Reply:
I’m still using relative `gpg`, not a full path.

@neverpanic @cjones051073 Would one of you please chime in with your opinion on the Right Thing To Do™ here?

Reply:
I added this as an option `gpg_verify.gpg`.